diffusion/models/models.py

```diff
@@ -86,20 +94,50 @@ def stable_diffusion_2(
         if isinstance(metric, CLIPScore):
             metric.requires_grad_(False)
 
-    if pretrained:
-        unet = UNet2DConditionModel.from_pretrained(model_name, subfolder='unet')
-    else:
-        config = PretrainedConfig.get_config_dict(model_name, subfolder='unet')
-        unet = UNet2DConditionModel(**config[0])
-
+    # Make the text encoder
     if encode_latents_in_fp16:
-        vae = AutoencoderKL.from_pretrained(model_name, subfolder='vae', torch_dtype=torch.float16)
         text_encoder = CLIPTextModel.from_pretrained(model_name, subfolder='text_encoder', torch_dtype=torch.float16)
     else:
-        vae = AutoencoderKL.from_pretrained(model_name, subfolder='vae')
         text_encoder = CLIPTextModel.from_pretrained(model_name, subfolder='text_encoder')
-
     tokenizer = CLIPTokenizer.from_pretrained(model_name, subfolder='tokenizer')
+
+    # Make the autoencoder
+    if autoencoder_path is None:
+        # Use the pretrained vae
+        downsample_factor = 8
+        if encode_latents_in_fp16:
+            vae = AutoencoderKL.from_pretrained(model_name, subfolder='vae', torch_dtype=torch.float16)
+        else:
+            vae = AutoencoderKL.from_pretrained(model_name, subfolder='vae')
+        if latent_scale is None:
+            # Default to the SD1/SD2 latent scale
+            latent_scale = 0.18215
+    else:
```

Comment:
Do the weights need to specified as `torch_dtype=torch.float16` somewhere for `encode_latents_in_fp16`?

Reply:
Added an option to load custom autoencoder weights in fp16 for when this option is enabled.